### PR TITLE
Integrating pychess python package

### PR DESCRIPTION
**Pull request by chelsea0x3b**
So this is a pretty big rewrite, so sorry for the awful diff. This PR will make it a ton easier to integrate a chess engine (like stockfish), because the python chess package has UCI chess engine interface built in. I can integrate a chess engine in another PR.

Here are the notable changes:

* Removing all chess specific code, as the chess library handles that now.
* Adding simple GPIO wrapper class called Magnet.
* Adding MotorMover class to handle moving the motors. This keeps track of the position instead of always moving back to 0,0.
* Input of moves is now done by reading in a UCI string. Promotion is now part of the move specifically (e.g. a7a8q moves from a7 to a8 and promotes to a queen).

I can run a regular chess game when commenting out the motor & magnet code, so I'm fairly confident in that. I'm not sure about the magnet & motor code though... I can't see anything obviously wrong, but you'll have to test it and let me know. :)

### CODE REVIEW

File: Wizard_chess.py [hunk 1]
```diff
@@ -1,728 +1,308 @@
...
+        """
+        GPIO.setup(7, GPIO.OUT)
+
+    @staticmethod
+    def grab():
+        """
+        Grab a piece with the magnet
+        :return: none
+        """
+        GPIO.output(7, 0)
```
Comment (coreylowman) [inline Wizard_chess.py]: I think these values are correct (i.e. 0 means move the magnet up to grab a piece and 1 means move the magnet down to release a piece), but please check :)

File: Wizard_chess.py [hunk 2]
```diff
@@ -1,728 +1,308 @@
...
+
+    def _compute_dist_and_rot_dir(self, new_value, old_value):
+        """
+        Calculates the distance and rotation that the motor has to travel for going to a new location
+        :param new_value: the new location
+        :param old_value: the old location
+        :return: (distance, rotation_direction), where rotation_direction is either Motor.CLOCKWISE or Motor.ANTICLOCKWISE
+        """
+        distance = self.unit_dist * abs(old_value - new_value)
+        rotation_direction = Motor.CLOCKWISE if new_value > old_value else Motor.ANTICLOCKWISE
```
Comment (coreylowman) [inline Wizard_chess.py]: I'm not quite sure if these rotation directions are correct, so if the pieces are moving the opposite of what they should be, its probably this line.

**Comment by bfentiman** [comment]
Thanks for the pull request. I will test it with the physical machine and update you so that you know what ends up happening with it. 👍

**Comment by chelsea0x3b** [comment]
Sounds good - note that playing against the computer will require you to download the stockfish executable (or some other UCI chess engine executable), and then modify the path to that executable on [Line 318](https://github.com/bfentiman/wizardchess/blob/master/Wizard_chess.py#L318)

**Pull request merged**